client/app/states/designer/blueprints/editor/canvasCtrl.js

```diff
@@ -53,9 +55,60 @@
     };
 
     $scope.addNewNode = function(newNode) {
-      $scope.chartViewModel.addNode(newNode);
+      if (!newNode.type || newNode.type !== 'new-generic') {
+        getTags(newNode.id).then(function(tags) {
+          newNode.tags = tags;
+          $scope.chartViewModel.addNode(newNode);
+        }, function() {
+          $scope.chartViewModel.addNode(newNode);
+        });
+      } else {
+        newNode.tags = [];
+        $scope.chartViewModel.addNode(newNode);
+      }
     };
 
+    function getTags(id) {
+      var deferred = $q.defer();
+
+      var attributes = ['categorization', 'category.id', 'category.single_value'];
+      var options = {
+        expand: 'resources',
+        attributes: attributes,
+      };
+
+      var collection = "service_templates" + "\/" + id + "\/" + 'tags';
```

Comment:
@dtaylor113 ignore this, it breaks things but shouldn't working on what i did wrong presently, sorry :-/